resources/bears-example.json

```diff
@@ -0,0 +1,127 @@
+{
+  "type": "failingPassing",
+  "repository": {
+    "name": "bowenli86/flink",
+    "githubId": 122434,
+    "url": "http://github.com/bowenli86/flink/",
+    "fork": true,
+    "isPR": true,
+    "PRID": 12,
+    "original": {
+      "name": "another/flink",
+      "githubId": 122435,
+      "url": "http://github.com/another/flink/"
+    }
+  },
+  "builds": {
+    "buggyBuild": {
+      "id": 122232233,
+      "url": "http://travis-ci.org/bowenli86/flink/builds/362501329",
+      "date": "Apr 5, 2018 6:12:24 AM"
+    },
+    "fixerBuild": {
+      "id": 122232233,
+      "url": "http://travis-ci.org/bowenli86/flink/builds/362501329",
+      "date": "Apr 5, 2018 6:12:24 AM"
+    }
+  },
+  "commits": {
+    "buggyBuild": {
+      "repoName": "bowenli86/flink",
+      "branchName": "master",
+      "sha": "f6940a0bf0fd5ff487a63ffad29e4dc9cd7a970c",
+      "url": "http://github.com/bowenli86/flink/commit/f6940a0bf0fd5ff487a63ffad29e4dc9cd7a970c",
+      "date": "Apr 5, 2018 6:12:36 AM"
+    },
+    "fixerBuild": {
+      "repoName": "bowenli86/flink",
+      "branchName": "master",
+      "sha": "f6940a0bf0fd5ff487a63ffad29e4dc9cd7a970c",
+      "url": "http://github.com/bowenli86/flink/commit/f6940a0bf0fd5ff487a63ffad29e4dc9cd7a970c",
+      "date": "Apr 5, 2018 6:12:36 AM"
+    },
+    "buggyBuildForkRepo": {
+      "repoName": "another/flink",
+      "branchName": "myPR",
+      "sha": "f6940a0bf0fd5ff487a63ffad29e4dc9cd7a970c",
+      "url": "http://github.com/bowenli86/flink/commit/f6940a0bf0fd5ff487a63ffad29e4dc9cd7a970c",
+      "date": "Apr 5, 2018 6:12:36 AM"
+    },
+    "buggyBuildBaseRepo": {
+      "repoName": "bowenli86/flink",
+      "branchName": "master",
+      "sha": "f6940a0bf0fd5ff487a63ffad29e4dc9cd7a970c",
+      "url": "http://github.com/bowenli86/flink/commit/f6940a0bf0fd5ff487a63ffad29e4dc9cd7a970c",
+      "date": "Apr 5, 2018 6:12:36 AM"
+    },
+    "fixerBuildForkRepo": {
+      "repoName": "another/flink",
+      "branchName": "myPR",
+      "sha": "f6940a0bf0fd5ff487a63ffad29e4dc9cd7a970c",
+      "url": "http://github.com/bowenli86/flink/commit/f6940a0bf0fd5ff487a63ffad29e4dc9cd7a970c",
+      "date": "Apr 5, 2018 6:12:36 AM"
+    },
+    "fixerBuildBaseRepo": {
+      "repoName": "bowenli86/flink",
+      "branchName": "master",
+      "sha": "f6940a0bf0fd5ff487a63ffad29e4dc9cd7a970c",
+      "url": "http://github.com/bowenli86/flink/commit/f6940a0bf0fd5ff487a63ffad29e4dc9cd7a970c",
+      "date": "Apr 5, 2018 6:12:36 AM"
+    }
+  },
+  "reproductionDate": "Apr 5, 2018 6:12:24 AM",
+  "tests": {
+    "overall": {
+      "numberExecuted": 363,
+      "numberFailing": 19,
+      "numberErroring": 1,
+      "numberSkipped": 0,
+      "numberPassing": 343
+    },
+    "failingModule": "flink-tests",
+    "failingClasses": [
+      {
+        "testClass": "org.apache.flink.util.FileUtilsTest",
```

Comment:
Then it would look something like that, which I find far more difficult to read:

```json
{
        "testClass": "org.apache.flink.util.FileUtilsTest",
        "failingMethods": [
          {
            "testMethod": "testOne",
            "failures": {
                "name": "java.lang.AssertionError",
                "isError": false,
              }
          },
          {
            "testMethod": "testTwo",
            "failures": {
                "name": "java.lang.AssertionError",
                "isError": false,
              }
          },
          {
            "testMethod": "test3",
            "failures": {
              "name": "java.lang.AssertionError",
              "isError": false,
            }
          },
          {
            "testMethod": "test4",
            "failures": {
              "name": "java.lang.AssertionError",
              "isError": false,
            }
          },
          {
            "testMethod": "test5",
            "failures": {
              "name": "java.lang.AssertionError",
              "isError": false,
            }
          },
          {
            "testMethod": "test6",
            "failures": {
              "name": "java.lang.AssertionError",
              "isError": false,
            }
          }
          ,{
            "testMethod": "test7",
            "failures": {
              "name": "java.lang.AssertionError",
              "isError": false,
            }
          }
        ]
      }
```

Reply:
And if we put only failingMethod but with fully-qualified name?

Reply:
It's the same problem: a failing method as only one failure, so we'll get a very long list of failing methods with their failures. And if it's only assertion errors, we'll see AssertionErrors everywhere.

So the question here: do you think you'll need to analyze for each test methods the failure? Or do you need more general metrics about the kind and number of failures? I can also display a list of failing methods without their attached failure.

Reply:
I can't say for you a real case I will need the detailed information, but I think it should be better to have it and to delete in the future if this is useless than to not have it and to be necessary.

To be honest, I don't think I will analyze the .json file directly (maybe only to check very small things quickly), but I will use bears-usage or the ongoing web interface instead.

Reply:
ok then I propose a middle way: to provide in overallMetrics the list of failures with number of occurences on the whole test suite, which also define if the test was an error or not. And then a detailed list of testclasses and methods with the obtained failure. It would look like that:

```
overallMetrics: {
  failures: [
     {
         className: "java.lang.AssertionError",
         isError: false,
         occurences: 19
     },
     {
         className: "java.lang.NullPointerException",
         isError: true,
         occurences: 19
     }
  ]
},
metricsByClass: [
  {
     testClass: "org.acme.MyClass",
     failures: 1,
     errors: 1
  },
  {
     testClass: "org.acme.OtherClass",
     failures: 1,
     errors: 0
  }
],
failureDetails: [
  {
     testClass: "org.acme.MyClass",
     testMethod: "test1",
     failure: "java.lang.AssertionError"
  },
  {
     testClass: "org.acme.MyClass",
     testMethod: "test2",
     failure: "java.lang.NullPointerException"
  },
  {
     testClass: "org.acme.OtherClass",
     testMethod: "test1",
     failure: "java.lang.AssertionError"
  }
]
```

WDYT?

Reply:
It seems good. But don't you think we also need "isError" in each item in "failureDetails" to be really complete?

Reply:
I can put it yes. OK so let's go to this solution